### PR TITLE
Escape slashes in post array before to call `wp_update_post`

### PR DESCRIPTION
**Pull request by strategio**
https://onthegosystems.myjetbrains.com/youtrack/issue/wpmlcore-6774

### CODE REVIEW

**Comment by strategio** [comment]
@OnTheGoSystems/wpml review please.

File: src/class-wpml-gutenberg-integration.php [hunk 1]
```diff
@@ -120,9 +113,7 @@ public function string_translated(
 				$content .= $this->render_block( $block );
 			}
 
-			$this->sitepress->switch_lang( $lang );
-			wp_update_post( array( 'ID' => $translated_post_id, 'post_content' => $content ) );
-			$this->sitepress->switch_lang( null );
+			wpml_update_escaped_post( [ 'ID' => $translated_post_id, 'post_content' => $content ], $lang );
```
Comment (strategio) [inline src/class-wpml-gutenberg-integration.php:116]: With the new function `wpml_update_escaped_post` we can pass the language as a parameter so I could remove the `SitePress` dependency (here only for this purpose).

**Comment by strategio** [comment]
@OnTheGoSystems/wpml, I made some adjustments, please review!

**Review by kagg-design — DISMISSED** [review]
+1 @strategio

**Comment by strategio** [comment]
@OnTheGoSystems/wpml, one last review please :)

**Review by andreasciamanna — DISMISSED** [review]

**Pull request merged**